source/jormungandr/jormungandr/instance_manager.py

```diff
@@ -64,6 +64,10 @@ def instances_comparator(instance1, instance2):
     if jormun_bdd_instance1.is_free != jormun_bdd_instance2.is_free:
         return jormun_bdd_instance1.is_free - jormun_bdd_instance2.is_free
 
+    #Here we choose the instance with greater priority.
+    if jormun_bdd_instance1.priority > jormun_bdd_instance2.priority:
```

Comment:
hum I think you want to test first if they are different, then return the subtraction of the value